csharp_dotnetcore/16.Handling-Atachments/Handling_Attachments/AtachmentsBot.cs

```diff
@@ -0,0 +1,272 @@
+        /// <param name="cancellationToken" >(Optional) A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>>A <see cref="Task"/> representing the operation result of the Turn operation.</returns>
+        private static async Task SendWelcomeMessageAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            foreach (var member in turnContext.Activity.MembersAdded)
+            {
+                if (member.Id != turnContext.Activity.Recipient.Id)
+                {
+                    await turnContext.SendActivityAsync(
+                        $"Welcome to AttachmentsBot {member.Name}." +
+                        $" This bot will introduce you to Attachments." +
+                        $"  Please select an option",
+                        cancellationToken: cancellationToken);
+                    await DisplayOptionsAsync(turnContext, cancellationToken);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Given the input from the message activity the user sent, create the appropriate response.
+        /// </summary>
+        /// <param name="turnContext">Provides the <see cref="ITurnContext"/> for the turn of the bot.</param>
+        /// <returns>A <see cref="Activity"/> to send as a response to the user's input.</returns>
+        private static Activity ProcessInput(ITurnContext turnContext)
+        {
+            var activity = turnContext.Activity;
+            var reply = activity.CreateReply();
+
+            if (activity.Attachments != null && activity.Attachments.Any())
+            {
+                HandleIncomingAttachment(activity, reply);
+            }
+            else
+            {
+                HandleOutgoingAttachment(activity, reply);
+            }
+
+            return reply;
+        }
+
+        /// <summary>
+        /// This method responds with an <see cref="Activity"/> that demonstrates the use of the type
+        /// of attachment the user requested.
+        /// </summary>
+        private static void HandleOutgoingAttachment(IMessageActivity activity, IMessageActivity reply)
+        {
+            if (activity.Text.StartsWith("1"))
+            {
+                reply.Text = "This is an inline attachment";
+                reply.Attachments = new List<Attachment>() { GetInlineAttachment() };
+            }
+            else if (activity.Text.StartsWith("2"))
+            {
+                reply.Text = "This is an attachment from a https url";
+                reply.Attachments = new List<Attachment>() { GetInternetAttachment() };
+            }
+            else if (activity.Text.StartsWith("3"))
+            {
+                reply.Text = "This is an uploaded attachment";
+                reply.Attachments = new List<Attachment>() { GetUploadedAttachmentAsync(reply.ServiceUrl, reply.Conversation.Id).Result };
```

Comment:
>GetUploadedAttachmentAsync [](start = 61, length = 26)

Weird to see an Async method here, but no await. Please confirm this is correct, and either fix or add a comment.